Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -655,26 +655,55 @@ extension ResourceFileElements {
+
+        /// Handles the conversion of a `.copy` resource rule of SPM
+        ///
+        /// - Parameters:
+        ///   - resourceAbsolutePath: The absolute path of that resource
+        /// - Returns: A ProjectDescription.ResourceFileElement mapped from a `.copy` resource rule of SPM
+        func handleCopyResource(resourceAbsolutePath: AbsolutePath) -> ProjectDescription.ResourceFileElement {
+            return .folderReference(path: Path(resourceAbsolutePath.pathString))
         }
-        guard !resourcesPaths.isEmpty else { return nil }
 
-        return .init(
-            resources: resourcesPaths.map { absolutePath in
-                let absolutePathGlob = absolutePath.extension != nil ? absolutePath : absolutePath.appending(component: "**")
-                return .glob(
-                    pattern: Path(absolutePathGlob.pathString),
-                    excluding: excluding.map {
-                        let excludePath = path.appending(RelativePath($0))
-                        let excludeGlob = excludePath.extension != nil ? excludePath : excludePath.appending(component: "**")
-                        return Path(excludeGlob.pathString)
-                    }
-                )
+        /// Handles the conversion of a `.process` resource rule of SPM
+        ///
+        /// - Parameters:
+        ///   - resourceAbsolutePath: The absolute path of that resource
+        /// - Returns: A ProjectDescription.ResourceFileElement mapped from a `.process` resource rule of SPM
+        func handleProcessResource(resourceAbsolutePath: AbsolutePath) -> ProjectDescription.ResourceFileElement {
+
+            let absolutePathGlob = resourceAbsolutePath.extension != nil ? resourceAbsolutePath : resourceAbsolutePath.appending(component: "**")
+            return .glob(
+                pattern: Path(absolutePathGlob.pathString),
+                excluding: excluding.map {
+                    let excludePath = path.appending(RelativePath($0))
+                    let excludeGlob = excludePath.extension != nil ? excludePath : excludePath.appending(component: "**")
+                    return Path(excludeGlob.pathString)
+                }
+            )
+        }
+
+        var resourceFileElements: [ProjectDescription.ResourceFileElement] = resources.map {
+            let resourceAbsolutePath = path.appending(RelativePath($0.path))
+
+            switch $0.rule {
+            case .copy:
+                return handleCopyResource(resourceAbsolutePath: resourceAbsolutePath)
+            case .process:
+                return handleProcessResource(resourceAbsolutePath: resourceAbsolutePath)
             }
-        )
+        }
+
+        // Add default resources path if necessary
+        // They are handled like a `.process` rule
+        if sources == nil {
+            resourceFileElements += defaultResourcePaths(from: path).map { handleProcessResource(resourceAbsolutePath: $0) }
+        }
+
+        // Sanity check
```

Comment:
I would not call it a sanity check, as it being empty is perfectly normal

Reply:
Good point, changed wording